app/workers/concerns/dump_worker.rb

```diff
@@ -40,16 +46,14 @@ def dump_type
     "project_#{dump_target}_export"
   end
 
-  def temp_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}"
-  end
-
   def csv_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}.csv"
+    @csv_tempfile ||= Tempfile.new(['export', '.csv'])
+    @csv_tempfile.path
   end
 
   def gzip_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}.gz"
+    @gzip_tempfile ||= Tempfile.new(['export', '.gz'])
+    @gzip_tempfile.path
   end
 
   def project_file_path
```

Comment:
I'm actually contemplating getting rid of all of that, and just using the project id instead of logins and display names. Changing the replacement like you say means you could create "collisions" (not really collisions since the file name itself is a UUID, but you could get your file written in the same directory as another project which has a similar name but just difference in non-alphanum chars)